server/interaction/npc.go

```diff
@@ -0,0 +1,132 @@
+package interaction
+
+import (
+	"github.com/joshprzybyszewski/cribbage/game"
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+// NPCType is an enum specifying which type of NPCType
+type NPCType int
+
+// Dumb, Simple, and Calculated are supported
+const (
+	Dumb NPCType = iota
+	Simple
+	Calculated
+)
+
+var npcs = [...]npcPlayer{
```

Comment:
we don't need `...` in there I think?

Reply:
Got this idea from [here](https://github.com/golang/go/blob/master/src/go/token/token.go). It's basically a map from `int` to `npcPlayer` (i.e. an array where the keys are indices). I suppose it's more expressive to use a proper map though?

Reply:
refactored this out